packages/accounts/src/__tests__/fetch-accounts-test.ts

```diff
@@ -0,0 +1,154 @@
+        expect.assertions(2);
+
+        // Given a mock RPC client returning a mock account at a given address.
+        const address = '1111' as Address<'1111'>;
+        const rpc = getMockRpc({
+            [address]: <RpcAccount>{
+                data: ['somedata', 'base64'],
+                executable: false,
+                lamports: 1_000_000_000n,
+                owner: '9999',
+                rentEpoch: 42n,
+            },
+        });
+
+        // When we fetch that account using the fetchEncodedAccount function.
+        const account = await fetchEncodedAccount(rpc, address);
+
+        // Then we expect the following parsed encoded account to be returned.
+        account satisfies MaybeEncodedAccount;
+        expect(account).toStrictEqual({
+            address,
+            data: new Uint8Array([178, 137, 158, 117, 171, 90]),
+            executable: false,
+            exists: true,
+            lamports: 1_000_000_000n,
+            programAddress: '9999',
+            rentEpoch: 42n,
+        });
+
+        // And the getAccountInfo RPC method to have been called with the given address and an explicity base64 encoding.
+        expect(rpc.getAccountInfo).toHaveBeenCalledWith(address, { encoding: 'base64' });
+    });
+
+    it('fetches and parses a missing encoded account from an RPC', async () => {
+        expect.assertions(2);
+
+        // Given an address and a mock RPC that does not contain an account at that address.
+        const address = '1111' as Address<'1111'>;
+        const rpc = getMockRpc({});
+
+        // When we try to fetch the account at that address using the fetchEncodedAccount function.
+        const account = await fetchEncodedAccount(rpc, address);
+
+        // Then we expect the following non-existing account to be returned.
+        account satisfies MaybeEncodedAccount;
+        expect(account).toStrictEqual({ address, exists: false });
+
+        // And the getAccountInfo RPC method to have been called with the given address and an explicity base64 encoding.
+        expect(rpc.getAccountInfo).toHaveBeenCalledWith(address, { encoding: 'base64' });
+    });
+
+    it('freezes the returned account', async () => {
+        expect.assertions(1);
+
+        // Given a mock RPC client returning a mock account at a given address.
+        const address = '1111' as Address<'1111'>;
+        const rpc = getMockRpc({
+            [address]: <RpcAccount>{
+                data: ['somedata', 'base64'],
+                executable: false,
+                lamports: 1_000_000_000n,
+                owner: '9999',
+                rentEpoch: 42n,
+            },
+        });
+
+        // When we fetch that account using the fetchEncodedAccount function.
+        const account = await fetchEncodedAccount(rpc, address);
+
+        // Then we expect the returned account to be frozen.
+        expect(account).toBeFrozenObject();
+    });
+});
+
+describe('fetchEncodedAccounts', () => {
```

Comment:
I'm a bit confused about what you're suggesting here sorry. Are you saying no fetch helpers at all? Should we then export the parsing helpers? Also see my response to Joe about not having fetcher helpers.

Reply:
I was reacting to the plural helper. I'd do `fetchEncodedAccount()` singular, and then let people figure out how to batch them _in the transport_ if needed. Otherwise, just `Promise.all()` them.

Reply:
Intersting! So would the default RPC transport be doing that sort of magic where a batch of `getAccountInfo` get turned into a `getMultipleAccounts`?

Because, if not, we'd be making people send N RPC calls instead of one by default which makes no sense to me.

Reply:
> We could create a transport that examines both the getAccountInfo and getMultipleAccounts payloads and merge them together into efficient RPC chunks.

Yeah, 100%. We should do that, but at the transport level, not at the ‘everyone maintains their own cache’ level.

We can totally consider making this a default transport in `packages/library/`.